### PR TITLE
feat(cloud-sql/sql-server): update to V2 sample

### LINKED ISSUE #1471: CloudSQL: The SQL Server tests are failing
They are consistently failing, both in Windows and Linux (sample [CI output](https://source.cloud.google.com/results/invocations/c0d30f8b-04a5-46eb-af49-4abc1702016f/targets/github%2Fdotnet-docs-samples%2Fcloud-sql%2Fsql-server/tests))

Error here:
```
Kestrel failed to start.
NotSpecified: (:String) [], RemoteException
Unhandled Exception: System.Data.SqlClient.SqlException: A connection was successfully established with the server, but then an error occurred during the pre-login handshake. (provider: TCP Provider, error: 35 - An internal exception was caught) ---> System.IO.IOException: Unable to read data from the transport connection: Connection reset by peer. ---> System.Net.Sockets.SocketException: Connection reset by peer
   at System.Net.Sockets.NetworkStream.Read(Byte[] buffer, Int32 offset, Int32 size)
   --- End of inner exception stack trace ---
   at System.Net.Sockets.NetworkStream.Read(Byte[] buffer, Int32 offset, Int32 size)
   at System.Data.SqlClient.SNI.SNITCPHandle.Receive(SNIPacket& packet, Int32 timeoutInMilliseconds)
   --- End of inner exception stack trace ---
   at System.Data.SqlClient.SqlInternalConnectionTds..ctor(DbConnectionPoolIdentity identity, SqlConnectionString connectionOptions, SqlCredential credential, Object providerInfo, String newPassword, SecureString newSecurePassword, Boolean redirectedUserInstance, SqlConnectionString userConnectionOptions, SessionData reconnectSessionData, Boolean applyTransientFaultHandling)
   at System.Data.SqlClient.SqlConnectionFactory.CreateConnection(DbConnectionOptions options, DbConnectionPoolKey poolKey, Object poolGroupProviderInfo, DbConnectionPool pool, DbConnection owningConnection, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionFactory.CreatePooledConnection(DbConnectionPool pool, DbConnection owningObject, DbConnectionOptions options, DbConnectionPoolKey poolKey, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionPool.CreateObject(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
   at System.Data.ProviderBase.DbConnectionPool.UserCreateRequest(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, UInt32 waitForMultipleObjectsTimeout, Boolean allowCreate, Boolean onlyOneCheckConnection, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionFactory.TryGetConnection(DbConnection owningConnection, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal oldConnection, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionInternal.TryOpenConnectionInternal(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionClosed.TryOpenConnection(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.SqlClient.SqlConnection.TryOpen(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.Open()
   at CloudSql.StartupExtensions.<>c__DisplayClass0_0.<OpenWithRetry>b__0() in /tmpfs/src/github/dotnet-docs-samples/cloud-sql/sql-server/Startup.cs:line 90
   at Polly.Policy.<>c__DisplayClass108_0.<Execute>b__0(Context ctx, CancellationToken ct) in /_/src/Polly/Policy.ExecuteOverloads.cs:line 18
   at Polly.Policy.<>c__DisplayClass138_0.<Implementation>b__0(Context ctx, CancellationToken token) in /_/src/Polly/Policy.SyncNonGenericImplementation.cs:line 18
   at Polly.Retry.RetryEngine.Implementation[TResult](Func`3 action, Context context, CancellationToken cancellationToken, ExceptionPredicates shouldRetryExceptionPredicates, ResultPredicates`1 shouldRetryResultPredicates, Action`4 onRetry, Int32 permittedRetryCount, IEnumerable`1 sleepDurationsEnumerable, Func`4 sleepDurationProvider) in /_/src/Polly/Retry/RetryEngine.cs:line 35
   at Polly.Retry.RetryPolicy.Implementation[TResult](Func`3 action, Context context, CancellationToken cancellationToken) in /_/src/Polly/Retry/RetryPolicy.cs:line 35
   at Polly.Policy.Implementation(Action`2 action, Context context, CancellationToken cancellationToken) in /_/src/Polly/Policy.SyncNonGenericImplementation.cs:line 18
   at Polly.Policy.Execute(Action`2 action, Context context, CancellationToken cancellationToken) in /_/src/Polly/Policy.ExecuteOverloads.cs:line 73
   at CloudSql.StartupExtensions.OpenWithRetry(DbConnection connection) in /tmpfs/src/github/dotnet-docs-samples/cloud-sql/sql-server/Startup.cs:line 82
   at CloudSql.StartupExtensions.InitializeDatabase() in /tmpfs/src/github/dotnet-docs-samples/cloud-sql/sql-server/Startup.cs:line 98
   at CloudSql.Program.Main(String[] args) in /tmpfs/src/github/dotnet-docs-samples/cloud-sql/sql-server/Program.cs:line 37
BUILD FAILED
```

I'm skipping the tests in #1469 until they are fixed.

### PR DESCRIPTION
**Pull request by jsimonweb**
- Move TCP socket example into separate file
- Update region tags
- Update DB_HOST env. variable to be INSTANCE_HOST

### CODE REVIEW

File: cloud-sql/sql-server/ConnectTcp.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+{
+    public class SqlServerTcp
+    {
+        public static SqlConnectionStringBuilder NewSqlServerTCPConnectionString()
+        {
+            // Equivalent connection string:
+            // "User Id=<DB_USER>;Password=<DB_PASS>;Server=<INSTANCE_HOST>;Database=<DB_NAME>;"
+            var connectionString = new SqlConnectionStringBuilder()
+            {
+                // Remember - storing secrets in plain text is potentially unsafe. Consider using
```
Comment (enocom) [inline cloud-sql/sql-server/ConnectTcp.cs:31]: Let's use the same wording as in the Go sample here.

File: cloud-sql/sql-server/runTestsWithSqlServer.ps1 [hunk 2]
```diff
@@ -12,42 +12,39 @@
...
-#	Stop-Job $proxy
-#	Receive-Job $proxy -ErrorAction 'SilentlyContinue'
-#	Remove-Job $proxy
-#}
\ No newline at end of file
+} finally {
+	Stop-Job $proxy
+	Receive-Job $proxy -ErrorAction 'Continue'
+	Remove-Job $proxy
+}
```
Comment (enocom) [inline cloud-sql/sql-server/runTestsWithSqlServer.ps1:50]: nit: newline

File: cloud-sql/sql-server/runTestsWithSqlServer.ps1 [hunk 3]
```diff
@@ -12,42 +12,39 @@
 # License for the specific language governing permissions and limitations under
 # the License.
 
-
-# Skipping because of https://github.com/GoogleCloudPlatform/dotnet-docs-samples/issues/1471
```
Comment (enocom) [inline cloud-sql/sql-server/runTestsWithSqlServer.ps1:16]: Could we make this a separate PR?

**Review by enocom — CHANGES_REQUESTED** [review]

File: cloud-sql/sql-server/ConnectTcp.cs (hunk 1 above)
Comment (jsimonweb) [reply cloud-sql/sql-server/ConnectTcp.cs]: Done.

File: cloud-sql/sql-server/runTestsWithSqlServer.ps1 (hunk 2 above)
Comment (jsimonweb) [reply cloud-sql/sql-server/runTestsWithSqlServer.ps1]: N/A

File: cloud-sql/sql-server/runTestsWithSqlServer.ps1 (hunk 3 above)
Comment (jsimonweb) [reply cloud-sql/sql-server/runTestsWithSqlServer.ps1]: Updated as requested.

File: cloud-sql/sql-server/app.yaml [hunk 4]
```diff
@@ -9,7 +9,7 @@ env_variables:
     DB_USER: <your-username>
     DB_PASS: <your-password>
     DB_NAME: <your-database-name>
-    DB_HOST: cloudsql
+    INSTANCE_HOST: cloudsql
     
 beta_settings:
-    cloud_sql_instances: "your-project-id:us-central1:instance-name=tcp:1433"
\ No newline at end of file
+    cloud_sql_instances: "<project-name>:<region>:<instance-name>=tcp:1433"
```
Comment (enocom) [inline cloud-sql/sql-server/app.yaml:15]: Nit: newline

**Review by enocom — APPROVED** [review]

**Review by amanda-tarafa — APPROVED** [review]
LGTM
